semgrep_output_v0.atd

```diff
@@ -119,15 +119,31 @@ type core_match_extra
 type dataflow_trace
   <ocaml attr="deriving show">
   <python decorator="dataclass(frozen=True)"> = {
-  ?taint_source: location list option;
+  ?taint_source: taint_source option;
```

Comment:
why taint_source is an option?

Reply:
Based on feedback on the original PR, https://github.com/returntocorp/semgrep-interfaces/pull/9, I named this type `dataflow_trace` instead of `taint_trace` with the idea that it could be used to explain other kinds of dataflow that Semgrep performs. For example, we could explain how constant propagation leads to a finding. In that case it would not be explaining a taint mode finding, so there wouldn't be a taint source to include.

Reply:
We could also have kept it as an optional `taint_trace` and then add another optional `const_prop_trace` ?

Reply:
I think that would also have been a reasonable approach.